### ISSUE TITLE
newline-after-var bug with comment line

### ISSUE THREAD

**Issue by evilebottnawi** [eslint/eslint#2852]
Ignore comments
```javascript
/* eslint-disable camelcase */
var photoSwipe = new PhotoSwipe($pswp, PhotoSwipeUI_Default, items, options);
/* eslint-enable no-underscore-dangle */
```
I get error newline-after-var

**Comment by lo1tuma** [comment]
This is the expected behavior of this rule.

**Comment by evilebottnawi** [comment]
@lo1tuma newline-after-var not throw error if after statement followed comments

**Comment by lo1tuma** [comment]
As I said, this is the expected behavior of the rule, not a bug.

**Comment by evilebottnawi** [comment]
@lo1tuma may be set configurable options (newline-after-var) for comments lines?

**Comment by lo1tuma** [comment]
Just to make sure I understand you correctly: you want an additional option for `newline-after-var` which allows comments in the line after a variable declaration? Is this correct?

**Comment by evilebottnawi** [comment]
@lo1tuma yes

**Comment by gyandeeps** [comment]
By providing that option we are basically turning this rule for that line.
with that being said, you can actually disable the rule inline

```
// eslint-disable-line newline-after-var
```

**Comment by evilebottnawi** [comment]
@gyandeeps bad solution, many files, many comments

**Comment by gyandeeps** [comment]
Now if you are doing this thing in many files and many comments, then I would recommend to turn off the rule.

**Comment by evilebottnawi** [comment]
@gyandeeps i need this rule

**Comment by nzakas** [comment]
@evilebottnawi is this just referring to ESLint disable comments or all comments?

**Comment by gyandeeps** [comment]
@evilebottnawi Waiting for your response.

**Comment by goloroden** [comment]
Although I'm not @evilebottnawi, I'm going to answer here ;-)

I have the same problem, and at least for me this is a just an issue with ESLint comments (disabling / enabling a rule) around a `var`, `const` or `let`.

**Comment by nzakas** [comment]
How would you expect this to behave?

**Comment by goloroden** [comment]
IMHO the rule `newline-after-var` should ignore lines after `var` that contain `eslint-disable`.

**Comment by goloroden** [comment]
@nzakas Any thoughts on this?

**Comment by IanVS** [comment]
I understand the goal here is to avoid having to write:

```js
/* eslint-disable camelcase */
var foo_default = "hello";

/* eslint-enable camelcase */
```

But, simply treating the config comment as a newline seems to violate the spirit of the rule.

```js
/* eslint-disable camelcase */
var foo_default = "hello";
/* eslint-enable camelcase */
if (foo === x) { ...
```

I propose instead that newline-after-var treat all comments immediately after var statements as additional var statements, and enforce a blank line after:

```js
/* eslint-disable camelcase */
var foo_default = "hello";
/* eslint-enable camelcase */

if (foo === x) { ...
```

This might also be useful when temporarily commenting out vars (e.g. for testing/debugging purposes).

```js
var foo = "hello";
var bar = require("world");
// var bar= " world";

console.log(foo + bar));
```

**Comment by goloroden** [comment]
@IanVS I very much like your idea :-) … +1

**Comment by evilebottnawi** [comment]
@goloroden thanks for answer for me, sorry i am busy.
@IanVS good news, thanks.

**Comment by nzakas** [comment]
@IanVS very creative solution, I like it.
